### PR TITLE
added assert for #881

### LINKED ISSUE #881: check duplicates in incAndAziInBus
`incAndAziInBus` may contain duplicates. We should check for this since this probably causes singularities.

### PR DESCRIPTION
**Pull request by Mathadon**
@damienpicard this is ready to merge if the unit tests pass.
This closes #881

### CODE REVIEW

File: IDEAS/BoundaryConditions/SolarIrradiation/RadSolData.mo [hunk 1]
```diff
@@ -34,11 +34,18 @@ model RadSolData "Selects or generates correct solar data for this surface"
...
+          else 0 for i in 1:numIncAndAziInBus});
+  final parameter Boolean solDataInBus = numMatches==1
     "True if the {inc,azi} combination is found in incAndAziInBus" annotation(Evaluate=true);
-  final parameter Integer solDataIndex=sum( { if sum(abs({mod(incAndAziInBus[i,1],Modelica.Constants.pi*2),mod(incAndAziInBus[i,2],Modelica.Constants.pi*2)} -
-  {mod(inc,Modelica.Constants.pi*2),mod(azi,Modelica.Constants.pi*2)}))<0.01 then i else 0 for i in 1:numIncAndAziInBus})
+  final parameter Integer solDataIndex=
+    sum( {if     IDEAS.Utilities.Math.Functions.isAngle(incAndAziInBus[i,1],inc)
+             and IDEAS.Utilities.Math.Functions.isAngle(incAndAziInBus[i,2],azi)
+          then i
+          else 0 for i in 1:numIncAndAziInBus})
```
Comment (damienpicard) [inline IDEAS/BoundaryConditions/SolarIrradiation/RadSolData.mo]: The previous implementation allowed some numerical errors (it turned out it was necessary to have it). Why did you remove it?

File: IDEAS/BoundaryConditions/SolarIrradiation/RadSolData.mo [hunk 2]
```diff
@@ -60,7 +67,10 @@ protected
...
     annotation (Placement(transformation(extent={{-100,-70},{-80,-50}})));
 equation
-      assert( not useLinearisation or (useLinearisation and solDataInBus), "The solar data must come
+    assert(numMatches<=1, "In "+getInstanceName()+
+      ": The parameter sim.incAndAziInBus contains duplicates. 
+      This is not allowed. Remove the duplicate entry.");
+    assert( not useLinearisation or (useLinearisation and solDataInBus), "The solar data must come
       from the weabus when the model is linearised. Add the combination {inc,azi} = {"+String(inc)+","+String(azi)+"}
       to the parameter incAndAziInBus of the SimInfoManager.");
   connect(radSol.solBus, solBusDummy) annotation (Line(
```
Comment (damienpicard) [inline IDEAS/BoundaryConditions/SolarIrradiation/RadSolData.mo]: handy!

File: IDEAS/BoundaryConditions/SolarIrradiation/RadSolData.mo (hunk 1 above)
Comment (damienpicard) [reply IDEAS/BoundaryConditions/SolarIrradiation/RadSolData.mo]: Ok, you have modified the isAngle.

Comment (Mathadon) [reply IDEAS/BoundaryConditions/SolarIrradiation/RadSolData.mo]: I moved the previous implementation to the function `isAngle` to have a better readable code.
